CRM/Upgrade/Headless.php

```diff
@@ -38,8 +44,11 @@ public function run($enablePrint = TRUE) {
     CRM_Core_DAO::dropTriggers();
 
     // CRM-11156
-    $preUpgradeMessage = NULL;
-    $upgrade->setPreUpgradeMessage($preUpgradeMessage, $currentVer, $latestVer);
+    if (empty($this->preUpgradeMessage)) {
+      $preUpgradeMessage = NULL;
+      $upgrade->setPreUpgradeMessage($preUpgradeMessage, $currentVer, $latestVer);
```

Comment:
Ug - this is a bit nasty that it is a pass-by-ref huh? Doesn't look like it is trivial to unravel though

Reply:
It is pass by reference

Reply:
Yeah, it's also a confusing name... but there are 3 distinct structures which include variations on this signature (`CRM_Upgrade_Incremental_Base` and its subclasses; then also `CRM_Upgrade_Form`; then also `CRM_Upgrade_Incremental_General`).